polygon-digital-carbon/src/RetirementHandler.ts

```diff
@@ -103,21 +105,25 @@ export function handleVCUOMinted(event: VCUOMinted): void {
   let sender = loadOrCreateAccount(event.transaction.from)
   let senderAddress = event.transaction.from
 
-  saveRetire(
-    sender.id.concatI32(sender.totalRetirements),
-    projectAddress,
-    ZERO_ADDRESS,
-    'OTHER',
-    retireAmount,
-    event.params.sender,
-    '',
-    senderAddress,
-    '',
-    event.block.timestamp,
-    event.transaction.hash
-  )
-
-  incrementAccountRetirements(senderAddress)
+  // Do not increment retirements for C3T-JCS tokens as the retirement has already been counted in StartAsyncToken
+  let token = Token.load(projectAddress)
+  if (token !== null && !token.symbol.startsWith('C3T-JCS') && !token.symbol.startsWith('C3T-ECO')) {
```

Comment:
Isn't this going to get out of hand with PURO or other registries coming? Maybe should be a list we check against?

Reply:
This event is only for C3 async retires, which are currently only JCS and ECO. So Puro shouldn't ever interact with this handler. If there continue to be more registries with async retirements from the C3ProjectTokenFactory though then yes a more organized list would avoid confusion. This is the only event that currently has a conditional between async and non-async c3 retirements.